projects/oit_demo/OITDemoApplication.cpp

```diff
@@ -381,17 +414,34 @@ void OITDemoApp::UpdateGUI()
     if (ImGui::Begin("Parameters")) {
         ImGui::Combo("Algorithm", &mGuiParameters.algorithmDataIndex, mSupportedAlgorithmNames.data(), static_cast<int>(mSupportedAlgorithmNames.size()));
 
-        ImGui::SliderFloat("Opacity", &mGuiParameters.meshOpacity, 0.0f, 1.0f, "%.2f");
-        ImGui::Checkbox("Display background", &mGuiParameters.displayBackground);
-        if (mGuiParameters.displayBackground) {
+        ImGui::Separator();
+        ImGui::Text("Model");
+        const char* meshesChoices[] =
+            {
+                "Monkey",
+                "Horse",
+                "Megaphone",
+                "Cannon",
+            };
+        static_assert(IM_ARRAYSIZE(meshesChoices) == MESH_TYPES_COUNT, "Mesh types count mismatch");
```

Comment:
If I remember well, std::array would not allow more than MESH_TYPES_COUNT elements, but it would allow less. For example, you couldn't put 5 elements in std::array<int, 4>, but you could put only 3 and it would compile.
The static_assert ensures that the array has exactly the necessary number of elements, at compile time.
Does that make sense?

Reply:
> but you could put only 3 and it would compile.

😢 I'm sad. So yes,static_assert makes sense. Thanks!

Reply:
Yes, I know the feeling. It's pretty sad the C way is actually stronger than the most modern way haha.